File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        indexMap = {}
        indexMap[nums[0]] = 0
        list = []
        for idx in range(1, len(nums)):
            difference = target - nums[idx]
            if difference in indexMap:
                list.append(idx)
                list.append(indexMap[difference])
            indexMap[nums[idx]] = idx
        return list
        
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] tuple = new int[2];
        HashMap<Integer, Integer> indexMap = new HashMap();
        indexMap.put(new Integer(nums[0]), new Integer(0));
        for(int i = 1; i < nums.length; i++) {
            int difference = target - nums[i];
            if(indexMap.containsKey(difference)) {
                tuple[0] = i;
                tuple[1] = indexMap.get(difference);
            }
            indexMap.put(new Integer(nums[i]), new Integer(i));
        }
        return tuple;
    }
}